sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -795,7 +795,7 @@ object NullPropagation extends Rule[LogicalPlan] {
  */
 object FoldablePropagation extends Rule[LogicalPlan] {
   def apply(plan: LogicalPlan): LogicalPlan = {
-    CleanupAliases(propagateFoldables(plan)._1)
+    EnforceGroupingReferencesInAggregates(CleanupAliases(propagateFoldables(plan)._1))
```

Comment:
why is this change needed?

Reply:
Hmm, sorry, I don't think this is needed. I will revert this in the next commit...

Reply:
Fixed in https://github.com/apache/spark/pull/31913/commits/06224449062e2f1e97a419adf44de45dce9eeb6d.